android/src/main/kotlin/com/almoullim/background_location/BackgroundLocationService.kt

```diff
@@ -114,6 +115,19 @@ class BackgroundLocationService: MethodChannel.MethodCallHandler, PluginRegistry
         return 0
     }
 
+    private fun isLocationServiceRunning(): Int {
+        val manager: ActivityManager = context!!.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+        for (service in manager.getRunningServices(Integer.MAX_VALUE)) {
+            if (LocationUpdatesService::class.java.getName() == service.service.getClassName()) {
+                if (service.foreground)
+                    return 1
```

Comment:
why is this returning 1 or 0 instead of a boolean true/false?

Reply:
Fair question. The result object of android in flutter v1 which is what I had initially made this additional feature in (around June 2021) has an int as the flied returned where the swift version was a little more dynamic. This was changed shortly after but I had missed. That. Will update it to also be a boolean since that is all that would be required